flox/core.py

```diff
@@ -1294,6 +1295,9 @@ def dask_groupby_agg(
 ) -> tuple[DaskArray, tuple[np.ndarray | DaskArray]]:
     import dask.array
     from dask.array.core import slices_from_chunks
+    from xarray.core.parallelcompat import get_chunked_array_type
+
+    chunkmanager = get_chunked_array_type(array)
```

Comment:
Obviously this approach would introduce a dependency on xarray, which presumably is not desirable.

Reply:
I'm fine just having `dask_kwargs` and `cubed_kwargs` instead of all this complexity.

Reply:
I probably should have just done that in xarray itself :sweat_smile: